sdk/textanalytics/ai-text-analytics/src/transforms.ts

```diff
@@ -0,0 +1,346 @@
+  const assessment: SentenceAssessment | undefined =
+    sentences?.[assessmentIndex.sentence].assessments?.[assessmentIndex.assessment];
+  if (assessment !== undefined) {
+    return assessment;
+  } else {
+    throw new Error(`Pointer "${assessmentPtr}" is not a valid Assessment pointer`);
+  }
+}
+
+/**
+ * @internal
+ */
+export function toEntityLinkingResult(
+  documents: TextDocumentInput[],
+  results: GeneratedEntityLinkingResult
+): EntityLinkingResult[] {
+  return transformDocumentResults(documents, results);
+}
+
+/**
+ * @internal
+ */
+export function toKeyPhraseExtractionResult(
+  documents: TextDocumentInput[],
+  results: GeneratedKeyPhraseExtractionResult
+): KeyPhraseExtractionResult[] {
+  return transformDocumentResults(documents, results);
+}
+
+/**
+ * @internal
+ */
+export function toEntityRecognitionResult(
+  documents: TextDocumentInput[],
+  results: GeneratedEntityRecognitionResult
+): EntityRecognitionResult[] {
+  return transformDocumentResults(documents, results);
+}
+
+/**
+ * @internal
+ */
+export function transformActionResult(
+  actionName: AnalyzeActionName,
+  input: TextDocumentInput[] | LanguageDetectionInput[],
+  response: AnalyzeResponse
+):
+  | EntityLinkingResult[]
+  | EntityRecognitionResult[]
+  | KeyPhraseExtractionResult[]
+  | SentimentAnalysisResult[]
+  | LanguageDetectionResult[] {
+  switch (response.kind) {
+    case "EntityLinkingResults": {
+      return toEntityLinkingResult(input, (response as EntityLinkingTaskResult).results);
+    }
+    case "EntityRecognitionResults": {
+      return toEntityRecognitionResult(input, (response as EntitiesTaskResult).results);
+    }
+    case "KeyPhraseExtractionResults": {
+      return toKeyPhraseExtractionResult(input, (response as KeyPhraseTaskResult).results);
+    }
+    case "PiiEntityRecognitionResults": {
+      return toPiiEntityRecognitionResult(input, (response as PiiTaskResult).results);
+    }
+    case "SentimentAnalysisResults": {
+      return toSentimentAnalysisResult(input, (response as SentimentTaskResult).results);
+    }
+    case "LanguageDetectionResults": {
+      return toLanguageDetectionResult(input, (response as LanguageDetectionTaskResult).results);
+    }
+    default: {
+      throw new Error(`Unsupported action name: ${actionName}`);
+    }
+  }
+}
+
+function appendReadableErrorMessage(currentMessage: string, innerMessage: string): string {
+  let message = currentMessage;
+  if (message.slice(-1) !== ".") {
+    message = message + ".";
+  }
+  return message + " " + innerMessage;
+}
+
+/**
+ * @internal
+ * parses incoming errors from the service/
+ * @param error - the incoming error
+ */
+export function transformError(errorResponse: unknown): any {
+  const castErrorResponse = errorResponse as {
+    response: {
+      parsedBody?: ErrorResponse;
+    };
+    statusCode: number;
+  };
+  if (!castErrorResponse.response) {
+    throw errorResponse;
+  }
```

Comment:
NIT: "casting" technically refers to something TypeScript doesn't do. It allows "type assertions," but a "cast" refers to some change in code generation or interpretation of data at runtime. How about `strongErrorResponse`?